pkg/portableresources/backend/controller/createorupdateresource.go

```diff
@@ -119,15 +119,25 @@ func (c *CreateOrUpdateResource[P, T]) Run(ctx context.Context, req *ctrl.Reques
 			return ctrl.Result{}, err
 		}
 	}
-
 	if recipeDataModel.Recipe() != nil {
 		recipeDataModel.Recipe().DeploymentStatus = util.Success
 	}
+
+	datamodel := recipeDataModel.(rpv1.RadiusResourceModel)
+
+	// Set recipe details on the resource metadata used.
+	if recipeOutput != nil && recipeOutput.Status != nil {
```

Comment:
`recipeOutput.Status` should always be populated as long as a recipe is used for deployment, and ignoring this can cause bugs to slip in. So it would be better to error out if this happens, let me know what you think.